src/Sulu/Component/HttpCache/EventSubscriber/InvalidationSubscriber.php

```diff
@@ -57,25 +91,100 @@ public function __construct(
     public static function getSubscribedEvents()
     {
         return [
-            Events::PUBLISH => ['invalidateDocumentForPublishing', -512],
+            Events::PUBLISH => ['invalidateDocumentBeforePublishing', 600],
+            Events::REMOVE => ['invalidateDocumentBeforeRemoving', 600],
```

Comment:
I started to use powers of 2 for defining priorities, that leaves enough room to put any other application specific listeners in between.